### DISCUSSION TITLE
Option to remove annoying 'long queue lines' notifications.

### DISCUSSION THREAD

**Discussion by ocalhoun6** [OpenRCT2/OpenRCT2#17879] [Feature requests]
This notification:
![Diamond Heights 2022-08-21 16-02-01](https://user-images.githubusercontent.com/26573303/185815768-54b3819e-dcc3-46a4-bb56-23d7c950eb07.png)

It gets pretty annoying when I _know_ the lines are long, but I have queue line TVs and entertainers and I don't care. Especially when it keeps popping up about the same rides over and over again.

I know the notification is there in lieu of hard limits on queue length, but it does seem overly aggressive about notifying the player, and it gets very annoying when you *want* to have long lines.

Option 1 (And probably the best option): Add a checkbox in the options window that lets you select which alerts to allow or mute, specifically for this alert without muting other alerts that I might want to see.

Option 2: Give the notification only once per ride. This will greatly reduce the number of repeat notifications. (To avoid needing to save this information in the save file, once per ride *per time you're playing* is fine -- not a big deal if the notifications come up again after saving and reloading the park.)

Option 3: Increase the limits that trigger the notification. The queue lines it's complaining about in my park are _far_ less than 1000 guests long. I was getting notifications about a ride that had less than 100 people in the queue. (Doppelganger in Diamond Heights, when its -- quite short -- queues were full.) The notifications would be less frequent and less annoying if they were only triggered by rides with _extremely_ long queues (over 1000 guests in line would be a good limit), rather than being triggered for every ride with a moderately long queue.

Any (or all three!) of these solutions would be a great way to make these alerts less annoying and less prone to false positives.

**Comment by duncanspumpkin** [comment]
You mean like disable the news guest warnings like we already have?

**Reply by IntelOrca** [reply]
A lot of players don't know it exists.
The ability to disable types of notifications is a window that is accessed by clicking a button on the recent messages window.

**Reply by ocalhoun6** [reply]
I know it exists, but I don't want to disable _all_ guest warnings. I still want to get notified if guests are complaining about disgusting paths, getting lost, can't find bathrooms, etc.

But the 'long queue lines' notifications are very frequent and annoying, and they're popping up for rides that don't even have that long of a line. I was getting repeated notifications for Doppelganger in Diamond Heights, which has two queue lines, each only 6 tiles long.

**Reply by karst** [reply]
I am not sure if it fits under ride warnings (i believe so) or park warnings. I think it'd be appreciated if the warning would mention what ride has the issue as well.

>A lot of players don't know it exists.

I have brought this up before (#16021), that it might be smart moving notification options to the other options. A lot of people do not even realize a notification tab exists. The actual button is disabled by default and people don't realize you can click date/weather to open the tab as well. It's overall not very intuitive. Maybe there could be an additional button in the options settings that opens the notification setting window, or have the window work similar to how ride selection works in the object selector with tabs below the another tab. I fully agree to not clutter up the menu's, but I wouldn't call making settings more intuitive to be cluttering up a menu.

There's a few menu buttons we could use for it. For example the newspaper/statistics button or
![2390](https://user-images.githubusercontent.com/1516893/186031406-fd3978eb-2a0d-4782-8465-77c9d1e1f661.png) from Locomotion.

**Comment by ocalhoun6** [comment]
Found the code responsible in Peep.cpp, starting at line 1183:

```
    if (warningThrottle[7])
        --warningThrottle[7];
    else if (tooLongQueueCounter > PEEP_TOO_LONG_QUEUE_THRESHOLD && tooLongQueueCounter > inQueueCounter / 20)
    { // The amount of guests complaining about queue duration is at least 5% of the amount of queuing guests.
      // This includes guests who are no longer queuing.
        warningThrottle[7] = 4;
```

If nothing else, can we raise this threshold to something more like 10% or 20%?

Or increase the threshold value (guest.h, line 28) to something much higher than 25. I'd say something more like 500.

--------

Really, though, I think this is the wrong approach for triggering the warning entirely.

It shouldn't be peeps that trigger the warning, it should be _rides_ that trigger the warning. Have each ride periodically check the queue length and/or queue time (which are both already tracked), and if it's above a certain threshold (say, 1000 peeps in line), trigger the warning.

Then it will only trigger on *truly* excessive queues.
